test_journal.py

```diff
@@ -0,0 +1,279 @@
+
+
+def test_write_entry(db_session):
+    kwargs = {'title': "Test Title", 'text': "Test entry text"}
+    kwargs['session'] = db_session
+    # first, assert that there are no entries in the database:
+    assert db_session.query(journal.Entry).count() == 0
+    # now, create an entry using the 'write' class method
+    entry = journal.Entry.write(**kwargs)
+    # the entry we get back ought to be an instance of Entry
+    assert isinstance(entry, journal.Entry)
+    # id and created are generated automatically, but only on writing to
+    # the database
+    auto_fields = ['id', 'created']
+    for field in auto_fields:
+        assert getattr(entry, field, None) is None
+    # flush the session to "write" the data to the database
+    db_session.flush()
+    # now, we should have one entry:
+    assert db_session.query(journal.Entry).count() == 1
+    for field in kwargs:
+        if field != 'session':
+            assert getattr(entry, field, '') == kwargs[field]
+    # id and created should be set automatically upon writing to db:
+    for auto in ['id', 'created']:
+        assert getattr(entry, auto, None) is not None
+
+
+def test_entry_no_title_fails(db_session):
+    bad_data = {'text': 'test text'}
+    journal.Entry.write(session=db_session, **bad_data)
+    with pytest.raises(IntegrityError):
+        db_session.flush()
+
+
+def test_entry_no_text_fails(db_session):
+    bad_data = {'title': 'test title'}
+    journal.Entry.write(session=db_session, **bad_data)
+    with pytest.raises(IntegrityError):
+        db_session.flush()
+
+
+def test_read_entries_empty(db_session):
+    entries = journal.Entry.all()
+    assert len(entries) == 0
+
+
+def test_read_entries_one(db_session):
+    title_template = "Title {}"
+    text_template = "Entry Text {}"
+    # write three entries, with order clear in the title and text
+    for x in range(3):
+        journal.Entry.write(
+            title=title_template.format(x),
+            text=text_template.format(x),
+            session=db_session)
+        db_session.flush()
+    entries = journal.Entry.all()
+    assert len(entries) == 3
+    assert entries[0].title > entries[1].title > entries[2].title
+    for entry in entries:
+        assert isinstance(entry, journal.Entry)
+
+
+def test_empty_listing(app):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    expected = 'No entries here so far'
+    assert expected in actual
+
+
+def test_listing(app, entry):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    for field in ['title']:
+        expected = getattr(entry, field, 'absent')
+        assert expected in actual
+
+
+def test_about_view(app):
+    response = app.get('/about')
+    assert response.status_code == 200
+    actual = response.body
+    expected = '<h1>Learning Journal</h1>'
+    assert expected in actual
+
+
+def test_create_view_redirect_then_login(app):
```

Comment:
is the name of this test accurate?  Does it really test that accessing the create view sends you to login?